tests/integration/test_logcollector/test_location/test_location.py

```diff
@@ -0,0 +1,246 @@
+        else:
+            with open(file_location, 'w') as file:
+                file.write(' ')
+    yield
+
+    for file_location in files:
+        if os.path.exists(file_location):
+            os.remove(file_location)
+
+
+def test_location(get_local_internal_options, configure_local_internal_options, create_directory, create_files,
+                  get_configuration, configure_environment,
+                  restart_logcollector):
+    """Check if logcollector is running properly with the specified configuration.
+
+    Raises:
+        TimeoutError: If the expected callback is not generated.
+    """
+    file_type = get_configuration['metadata']['file_type']
+    files = get_configuration['metadata']['files']
+
+    for file_location in sorted(files):
+        if file_type == 'single_file':
+            log_callback = logcollector.callback_analyzing_file(file_location,
+                                                                prefix=prefix)
+
+            wazuh_log_monitor.start(timeout=60, callback=log_callback,
+                                    error_message="The expected 'Analyzing file' message has not been produced")
+        elif file_type == 'wildcard_file':
+            pattern = get_configuration['metadata']['location']
+            log_callback = logcollector.callback_match_pattern_file(pattern, file_location,
+                                                                    prefix=prefix)
+            wazuh_log_monitor.start(timeout=60, callback=log_callback,
+                                    error_message=f"The expected 'New file that matches the '{pattern}' "
+                                                  f"pattern: '{file_location}' message has not been produced")
+        elif file_type == 'non_existent_file':
+            log_callback = logcollector.callback_non_existent_file(file_location,
+                                                                   prefix=prefix)
+            wazuh_log_monitor.start(timeout=60, callback=log_callback,
+                                    error_message="The expected 'Could not open file' message has not been produced")
+        elif file_type == 'duplicated_file':
+            log_callback = logcollector.callback_duplicated_file(file_location,
+                                                                 prefix=prefix)
+            wazuh_log_monitor.start(timeout=60, callback=log_callback,
+                                    error_message=f"The expected 'Log file '{file_location}' is duplicated' "
+                                                  f"message has not been produced")
+        elif file_type == 'multiple_logs':
+            log_callback = logcollector.callback_file_limit(prefix=prefix)
+            wazuh_log_monitor.start(timeout=60, callback=log_callback,
+                                    error_message=f"The expected 'File limit has been reached' "
+                                                  f"message has not been produced")

```

Comment:
Please try to avoid hard-coded values such as those 60 seconds in timeout. Use logcollector pre-defined timeout vars instead.

Reply:
Fixed in https://github.com/wazuh/wazuh-qa/commit/0745ed3a26b428b592929695d4e10c00605c5dd4